pandas/core/indexes/datetimes.py

```diff
@@ -916,6 +917,10 @@ def astype(self, dtype, copy=True):
             elif copy is True:
                 return self.copy()
             return self
+        elif is_categorical_dtype(dtype):
+            from pandas.core.indexes.category import CategoricalIndex
```

Comment:
pattern is to import ``CI`` at the top, but ok here too

e.g. ``from pandas.core.indexes.category import CategoricalIndex`` should work